api/src/main/java/org/pmiops/workbench/exfiltration/EgressEventServiceImpl.java

```diff
@@ -73,22 +84,62 @@ public void handleEvent(SumologicEgressEvent event) {
     } else {
       workspaceNamespace = dbWorkspaceMaybe.get().getWorkspaceNamespace();
     }
+    List<Optional<DbUser>> egressUsers = findEgressUsers(event, dbWorkspaceMaybe);
+
+    // This would happen if we receive a second GKE Egress for the same event when the app is being
+    // STOPPED,
+    // we just audit in this case since we can't know the user
+    if (egressUsers.isEmpty()) {
+      this.egressEventAuditor.fireEgressEvent(event);
+    }
+
+    for (Optional<DbUser> egressUserMaybe : egressUsers) {
+      logger.warning(
+          String.format(
+              "Received an egress event from workspace namespace %s, googleProject %s (%.2fMiB, VM name %s)",
+              workspaceNamespace, event.getProjectName(), event.getEgressMib(), event.getVmName()));
+      this.egressEventAuditor.fireEgressEventForUser(event, egressUserMaybe.orElse(null));
+
+      Optional<DbEgressEvent> maybeEvent =
+          this.maybePersistEgressEvent(event, egressUserMaybe, dbWorkspaceMaybe);
+      maybeEvent.ifPresent(e -> taskQueueService.pushEgressEventTask(e.getEgressEventId()));
+    }
+  }
+
+  @NotNull
+  private List<Optional<DbUser>> findEgressUsers(
+      SumologicEgressEvent event, Optional<DbWorkspace> dbWorkspace) {
+
+    // This case is when the Egress is from a GCE cluster.
+    if (StringUtils.isNotEmpty(event.getVmPrefix())
+        && event.getVmPrefix().startsWith("all-of-us")) {
```

Comment:
Can `getVmPrefix` returns null? Shall we add `isNotEmpty` check?

Reply:
it's in the if condition already, are you suggesting something else?

Reply:
ah, `StringUtils.isNotEmpty` has null check. Sorry for the confusion.
(btw, we also use guava a lot in RW https://cs.github.com/?scope=org%3Aall-of-us&scopeName=all-of-us&q=isNullOREmpty, 1st time seeing this similar method from a different library)

Reply:
interesting, I didn't know guava has a similar method. I usually use commons lang. Do you have a strong preference to update it?